File: frontend/src/lib/api/schemas/projects.ts
import { z } from "zod";
import { ProjectStatus } from "@/types/enums";

const projectStatusSchema = z.enum([
  ProjectStatus.IDEA,
  ProjectStatus.IN_PROGRESS,
  ProjectStatus.IN_REVIEW,
  ProjectStatus.COMPLETED,
  ProjectStatus.ARCHIVED,
]);

export const projectReadSchema = z.object({
  id: z.string().uuid(),
  channel_id: z.string().uuid(),
  trend_id: z.string().uuid().nullable(),
  created_by_user_id: z.string().uuid().nullable(),
  title: z.string(),
  status: projectStatusSchema,
  budget_usd: z.string().nullable(),
});
export type ProjectRead = z.infer<typeof projectReadSchema>;

export type ProjectCreateRequest = {
  title: string;
  trend_id?: string | null;
  budget_usd?: string | null;
};

export type ProjectUpdateRequest = {
  title?: string | null;
  budget_usd?: string | null;
};

export type ProjectStatusUpdateRequest = {
  status: ProjectStatus;
};

File: backend/src/ytforge/interfaces/api/schemas/projects.py
from __future__ import annotations

import uuid
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from ytforge.domain.enums import ProjectStatus


class ProjectCreateRequest(BaseModel):
    title: str = Field(min_length=1, max_length=500)
    trend_id: uuid.UUID | None = None
    budget_usd: Decimal | None = None


class ProjectUpdateRequest(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=500)
    budget_usd: Decimal | None = None


class ProjectStatusUpdateRequest(BaseModel):
    status: ProjectStatus


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    channel_id: uuid.UUID
    trend_id: uuid.UUID | None
    created_by_user_id: uuid.UUID | None
    title: str
    status: ProjectStatus
    budget_usd: Decimal | None
